src/commands/cmd_key.cc

```diff
@@ -424,6 +424,115 @@ class CommandCopy : public Commander {
   bool replace_ = false;
 };
 
+template <bool ReadOnly>
+class CommandSort : public Commander {
+ public:
+  Status Parse(const std::vector<std::string> &args) override {
+    CommandParser parser(args, 2);
+    while (parser.Good()) {
+      if (parser.EatEqICase("BY")) {
+        sort_argument_.sortby = GET_OR_RET(parser.TakeStr());
```

Comment:
Should we check `sortby.empty` here?

Reply:
I don't think it's necessary, redis allows receiving `""` as the value of `BY`, but this reminds me that I can add some test cases, thanks

Reply:
I mean, would "BY a BY b*" sequence making `sortby` has conflict stats (both dontsort and have properties)

Reply:
I try to test behavior like this in redis
```
127.0.0.1:6379> lpush uid 1 3 2
(integer) 3
127.0.0.1:6379> mset uid_rank_1 3 uid_rank_2 2 uid_rank_3 1
OK
127.0.0.1:6379> mset uid_level_1 1 uid_level_2 2 uid_level_3 3
OK
127.0.0.1:6379> sort uid by uid_rank_*
1) "3"
twenty two"
3) "1"
127.0.0.1:6379> sort uid by uid_level_*
1) "1"
twenty two"
3) "3"
127.0.0.1:6379> sort uid by uid_level_* by uid_rank_*
1) "3"
twenty two"
3) "1"
127.0.0.1:6379> sort uid by uid_rank_* by uid_level_*
1) "1"
twenty two"
3) "3"
127.0.0.1:6379> sort uid by dontsort by uid_level_*
1) "2"
twenty three"
3) "1"
127.0.0.1:6379> sort uid by dontsort by uid_rank_*
1) "2"
twenty three"
3) "1"
127.0.0.1:6379> sort uid by uid_level_* by dontsort
1) "2"
twenty three"
3) "1"
127.0.0.1:6379> sort uid by uid_rank_* by dontsort
1) "2"
twenty three"
3) "1"
```
It seems that when multiple BY appears, the next value will prevail. If dontsort is included, the output will be according to dontsort. 
Using multiple BYs does not seem like normal behavior, what do you think would be better? Align with redis or check?

Reply:
I think we can add a check and return invalid, following the redis would be a bit weird